telemetry/telemetry.proto

```diff
@@ -4,6 +4,76 @@ import 'nanopb.proto';
 
 option java_outer_classname="TelemetryProto";
 
-message Telemetry {
-  optional int32 timestamp_ms = 1;
+//A DeviceInfoMessage should only be sent if a request is received from the telemetry viewer (ex. if the
+//		viewer sees a device it hasn't registered yet)
+//Otherwise, the viewer will just reference its config files (stored offline) for this information
+//TODO: Make a better model for determining anomalies (a machine learning algorithm perhaps?)
+message DeviceInfoMessage { //send the device ID with its corresponding information
+	message DeviceMeasurementInfo {
+		enum READING_TYPE {
+			VOLTAGE = 1;
+			CURRENT = 2;
+			TEMPERATURE = 3;
+			MCC = 4;
+			DRIVER_CONTROL = 5; //i.e. pedal board
+		}
+		optional READING_TYPE readingType = 1;
+		optional string units = 2 [(nanopb).max_size = 4]; //max of four letters (mV, A, degC, mph, etc.)
+		optional uint32 minSensorLimit = 3;
+		optional uint32 maxSensorLimit = 4;
+		optional uint32 minErrorBound = 5; //if a reading < minErrorBound -> send error message
+		optional uint32 maxErrorBound = 6; //if a reading > maxErrorBound -> send error message
+		optional uint32 resolution = 7;
+		optional uint32 accuracy = 8;
+	}
+
+	message DeviceDelayInfo {
+		optional uint32 expectedSampleDelay = 1; //expected time between sample readings
+		optional uint32 longestAcceptedSampleDelay = 2; //if exceeded, indicates a probable lack of connection
+	}
+
+	optional uint32 id = 1;
+	optional string name = 2 [(nanopb).max_size = 32];
+	optional uint32 numAggregateSamples = 3; //= numSamples in StatisticalAggregate (below)
```

Comment:
This could be dynamic (per report, not once in the device info)

Reply:
Just the numAggregateSamples? I'm thinking I can just take this out of DeviceInfoMessage because it's already being sent in each TelemetryMessage in the StatisticalMessage

Reply:
Yeah, just that. I think removal makes sense.